exist-core/src/main/java/org/exist/util/SaxonConfigurationHolder.java

```diff
@@ -0,0 +1,218 @@
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package org.exist.util;
+
+import com.evolvedbinary.j8fu.lazy.AtomicLazyVal;
+import net.sf.saxon.lib.Feature;
+import net.sf.saxon.s9api.Processor;
+import net.sf.saxon.trans.XPathException;
+import org.exist.storage.BrokerPool;
+import org.jline.utils.Log;
+
+import javax.xml.transform.stream.StreamSource;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public final class SaxonConfigurationHolder {
+
+  public final static String CONFIGURATION_ELEMENT_NAME = "saxon";
+  public final static String CONFIGURATION_FILE_ATTRIBUTE = "configuration-file";
+  public final static String CONFIGURATION_FILE_PROPERTY = "saxon.configuration";
+  private final static String DEFAULT_SAXON_CONFIG_FILE = "saxon-config.xml";
+
+  /**
+   * Maintain a separate, singleton Saxon configuration for each broker pool
+   */
+  private static final Map<BrokerPool, SaxonConfigurationHolder> perBroker = new HashMap<>();
+
+  private final BrokerPool brokerPool;
+  /**
+   * Load the Saxon configuration for this broker pool on demand
+   */
+  private final AtomicLazyVal<net.sf.saxon.Configuration> saxonConfiguration = new AtomicLazyVal<>(this::loadConfiguration);
+
+  /**
+   * Create a Saxon processor from the Saxon configuration on demand
+   */
+  private final AtomicLazyVal<Processor> saxonProcessor = new AtomicLazyVal<>(this::createProcessor);
+
+  /**
+   * Holds the Saxon configuration related to a broker pool
+   * Configuration elements are loaded on demand,
+   * initially this just holds a reference to the owning {@link BrokerPool}
+   *
+   * @param brokerPool which owns this Saxon configuration
+   */
+  private SaxonConfigurationHolder(final BrokerPool brokerPool) {
+    this.brokerPool = brokerPool;
+  }
+
+  /**
+   * Factory for the singleton Saxon configuration wrapper per broker pool
+   * @param brokerPool for which to fetch (and if necessary create) a Saxon configuration
+   *
+   * @return the associated Saxon configuration wrapper
+   */
+  public synchronized static SaxonConfigurationHolder GetHolderForBroker(final BrokerPool brokerPool) {
```

Comment:
Why did you choose to start the static method name with an uppercase letter?

Reply:
My brain seems to believe that is the standard. I can change it.

Reply:
Changed to camel.